core/templates/dev/head/domain/sidebar/SidebarStatusServiceSpec.js

```diff
@@ -0,0 +1,52 @@
+describe('SidebarStatusService', function() {
+  var SidebarStatusService;
+
+  beforeEach(module('oppia'));
+  beforeEach(inject(function($injector) {
+    SidebarStatusService = $injector.get('SidebarStatusService');
+  }));
+
+  it('should open the sidebar if it is not open', function() {
+    SidebarStatusService.openSidebar();
+    expect(SidebarStatusService.isSidebarShown()).toBe(true);
+  });
+
+  it('should not open the sidebar if it is open', function() {
+    SidebarStatusService.openSidebar();
+    SidebarStatusService.openSidebar();
+    expect(SidebarStatusService.isSidebarShown()).toBe(true);
+  });
+
+  it('should close the sidebar if it is open', function() {
+    SidebarStatusService.openSidebar();
+    SidebarStatusService.closeSidebar();
+    expect(SidebarStatusService.isSidebarShown()).toBe(false);
+  });
+
+  it('should not close the sidebar if it is not open', function() {
+    SidebarStatusService.closeSidebar();
+    expect(SidebarStatusService.isSidebarShown()).toBe(false);
+  });
+
+  it('should toggle sidebar to open', function(){
+    SidebarStatusService.toggleSidebar();
+    expect(SidebarStatusService.isSidebarShown()).toBe(true);
+  });
+
+  it('should toggle sidebar to close', function(){
+    SidebarStatusService.openSidebar();
+    SidebarStatusService.toggleSidebar();
```

Comment:
You can move this case with the previous one. Meaning

it('should toggle sidebar', function() {
.... // the case for toggle sidebar to open
SidebarStatusService.toggleSidebar();
expect(...).toBe(false);
});

Reply:
Yes! We could add both the toggle tests with the open and close tests. It'll be more feasable I guess.